pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -168,6 +143,68 @@ public class ModelInterpreter implements Serializable {
         c.call()
     }
 
+
+    def evaluateStage(Root root, Agent parentAgent, Stage thisStage, Throwable firstError, Stage parentStage = null) {
+        return {
+            try {
+                // NOTE - this will switch to script.stage in the future.
+                if (firstError != null) {
+                    Utils.logToTaskListener("Stage '${thisStage.name}' skipped due to earlier failure(s)")
+                    Utils.markStageSkippedForFailure(thisStage.name)
+                } else if (skipUnstable(root.options)) {
+                    Utils.logToTaskListener("Stage '${thisStage.name}' skipped due to earlier stage(s) marking the build as unstable")
+                    Utils.markStageSkippedForUnstable(thisStage.name)
+                } else {
+                    // While we run the top-level environment block after the top-level
+                    // agent, we do the reverse per-stage. Why? So that the per-stage
+                    // environment is populated before we evaluate any when condition,
+                    // and so that we don't go into a per-stage agent if the when condition
+                    // isn't satisfied.
+                    withEnvBlock(thisStage.getEnvVars(root, script, parentStage)) {
+                        if (evaluateWhen(thisStage.when)) {
+                            if (thisStage.parallelStages != null) {
+                                def parallelStages = [:]
+                                for (int i = 0; i < thisStage.parallelStages.stages.size(); i++) {
+                                    Stage parallelStage = thisStage.parallelStages.getStages().get(i)
```

Comment:
Can you please stick to one syntax type, either `parallelStages.stages` or `parallelStages.getStages()`. I would prefer `parallelStages.stages[i]` as well instead of `parallelStages.stages.get(i)`